### PR TITLE
feat(ingest): add option to specify source platform database in lookml ingestion

### PR DESCRIPTION
**Pull request by remisalmon**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable)

This PR adds the option to specify the source platform database in the `connection_to_platform_map` configuration in order to generate proper upstream urns and lineages to the source tables in DataHub.

The source platform(s) and associated database(s) are configured in https://<subdomain>.looker.com/admin/connections and the tables in the lkml files are in the format `schema.table`.

Also changed the sql tables names to lowercase because urns are case sensitive in DataHub and the sqlalchemy-based metadata ingestion generates lowercase urns (at least Snowflake does).

### CODE REVIEW

File: metadata-ingestion/README.md [hunk 1]
```diff
@@ -622,8 +622,8 @@ source:
   config:
     base_folder: /path/to/model/files # Where the *.model.lkml and *.view.lkml files are stored.
     connection_to_platform_map: # mapping between connection names in the model files to platform names.
-      my_snowflake_conn: snowflake
-    platform_name: looker_views # Optional, default is "looker_views"
+      connection_name: platform_name # map to platform_name or platform_name.database_name
+    platform_name: "looker" # optional, default is "looker"
```
Comment (hsheth2) [inline metadata-ingestion/README.md:626]: thanks! I must've missed this edit in a previous PR

File: metadata-ingestion/README.md [hunk 2]
```diff
@@ -622,8 +622,8 @@ source:
   config:
     base_folder: /path/to/model/files # Where the *.model.lkml and *.view.lkml files are stored.
     connection_to_platform_map: # mapping between connection names in the model files to platform names.
-      my_snowflake_conn: snowflake
-    platform_name: looker_views # Optional, default is "looker_views"
+      connection_name: platform_name # map to platform_name or platform_name.database_name
```
Comment (hsheth2) [inline metadata-ingestion/README.md:625]: might be good to add two example items here, so it's clear that `connection_name` isn't a special key but rather an example placeholder

File: metadata-ingestion/src/datahub/ingestion/source/lookml.py [hunk 3]
```diff
@@ -382,7 +382,15 @@ def _load_model(self, path: str) -> LookerModel:
 
     def _construct_datalineage_urn(self, sql_table_name: str, connection: str) -> str:
         platform = self._get_platform_based_on_connection(connection)
-        return f"urn:li:dataset:(urn:li:dataPlatform:{platform},{sql_table_name},{self.source_config.env})"
+
+        if "." in platform:
+            platform_name, database_name = platform.lower().split(".")
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/lookml.py:387]: I think we want `maxsplit=1` here 
```suggestion
            platform_name, database_name = platform.lower().split(".", 1)
```

**Review by hsheth2 — CHANGES_REQUESTED** [review]
couple minor nits, but otherwise looks good!

**Comment by remisalmon** [comment]
@hsheth2 thanks for the quick review, last PR should resolve your comments.

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**